indico/modules/auth/controllers.py

```diff
@@ -117,18 +119,32 @@ def _process(self):
         return multipass.logout(request.args.get('next') or url_for('misc.index'), clear_session=True)
 
 
-def _send_confirmation(email, salt, endpoint, template, template_args=None, url_args=None, data=None):
+def _send_confirmation(email, salt, endpoint, template, template_args=None, url_args=None, data=None, flashed_msg=None):
     template_args = template_args or {}
     url_args = url_args or {}
     token = secure_serializer.dumps(data or email, salt=salt)
     url = url_for(endpoint, token=token, _external=True, _secure=True, **url_args)
     template_module = get_template_module(template, email=email, url=url, **template_args)
     GenericMailer.send(make_email(email, template=template_module))
-    flash(_('We have sent you a verification email. Please check your mailbox within the next hour and open '
-            'the link in that email.'))
+
+    if not flashed_msg:
```

Comment:
I would always flash the msg in the function calling `_send_confirmation` instead of sometimes flashing a local msg and sometimes one from outside.